client/src/App.js

```diff
@@ -29,6 +28,8 @@ const socketOccupancies = new Observable(notify => {
   });
 }).pipe(map(payload => ({ type: "setOccupancy", payload })));
 
+agent.addFilter(({ action }) => store.dispatch(action));
+agent.on("holdRoom", ({ action }) => socket.emit("holdRoom", action.payload));
```

Comment:
Best practice is to keep filter and renderer setup next together, unless for a good reason like deferring their setup.

Reply:
`addFilter` adds a filter for every action, whereas `filter` has a syntax more like `on`, where the first argument specifies a string, regex, or boolean test for whether to run it for each action.